web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.web.logging
+
+import com.ritense.valtimo.contract.annotation.SkipComponentScan
+import mu.KotlinLogging
+import org.springframework.boot.web.client.RestClientCustomizer
+import org.springframework.http.HttpRequest
+import org.springframework.http.client.ClientHttpRequestExecution
+import org.springframework.http.client.ClientHttpRequestInterceptor
+import org.springframework.http.client.ClientHttpResponse
+import org.springframework.stereotype.Component
+import org.springframework.web.client.HttpClientErrorException
+import org.springframework.web.client.RestClient
+import kotlin.text.Charsets.UTF_8
+
+@Component
+@SkipComponentScan
+class LoggingRestClientCustomizer : RestClientCustomizer, ClientHttpRequestInterceptor {
```

Comment:
Not super happy that both interfaces are used in a single class. Rather I would split them up. 1 RestClientCustomizer and 1 ClientHttpRequestInterceptor.

Reply:
hmm I kinda like it this way tbh

Reply:
Alright

Reply:
I think it's fine to combine them both in this case.